### PR TITLE
update libQGLViewer easyblock to take into account changes in the shared library names depending on Qt versions they are compiled with

### PR DESCRIPTION
**Pull request by satishskamath**
For the newer versions of libQGLviewer the qt versions are also included in the library names, therefore sanity check needs to be modified.

### CODE REVIEW

File: easybuild/easyblocks/l/libqglviewer.py [hunk 1]
```diff
@@ -49,10 +50,26 @@ def configure_step(self):
...
+                'files': [('lib/libQGLViewer.prl', 'lib64/libQGLViewer.prl'),
+                          ('lib/libQGLViewer.%s' % shlib_ext, 'lib64/libQGLViewer.%s' % shlib_ext)],
+                'dirs': ['include/QGLViewer'],
+            }
+        else:
+            addition = ''
+            for dep in self.cfg.dependencies():
+                if (dep['name'][0] == 'Q' and dep['name'][1] == 't'):
+                    addition = dep['name'].lower()
+                    addition = '-' + addition
```
Comment (Micket) [inline easybuild/easyblocks/l/libqglviewer.py:67]: I think this looks a bit cryptic and seems a bit fragile considering things like `QtPy` `Qtconsole`  etc.
I'd say just go with hardcoding Qt5 and 6 for now. We'll worry about generalizing it in 10 years when Qt7 comes.
```suggestion
            for qt_dep in ['Qt5', 'Qt6']:
                if get_software_root('Qt5'):
                    addition = qt_dep.lower()
                    break
            else:
                raise EasyBuildError("Missing Qt5 or Qt6 dependency")
```

**Review by Micket — CHANGES_REQUESTED** [review]

**Review by Minidixit — DISMISSED** [review]

File: easybuild/easyblocks/l/libqglviewer.py [hunk 2]
```diff
@@ -49,10 +52,29 @@ def configure_step(self):
...
+                    break
+            else:
+                raise EasyBuildError("Missing Qt5 or Qt6 dependency")
+            custom_paths = {
+                'files': [('lib/libQGLViewer'+addition+'.prl', 'lib64/libQGLViewer'+addition+'.prl'),
+                          ('lib/libQGLViewer'+addition+'.%s' % shlib_ext,
+                              'lib64/libQGLViewer'+addition+'.%s' % shlib_ext)],
+                'dirs': ['include/QGLViewer'],
+            }
```
Comment (Micket) [inline easybuild/easyblocks/l/libqglviewer.py:78]: Sorry for nitpicking here, but i think deduplicating it a bit like this is nicer, and `suffix` seems a bit more appropriate name
```suggestion
        if LooseVersion(self.version) < LooseVersion("2.8.0"):
            suffix = ''
        else:
            for dep in ['Qt5', 'Qt6']:
                if get_software_root(dep):
                    suffix = '-' + dep.lower()
                    break
            else:
                raise EasyBuildError("Missing Qt5 or Qt6 dependency")
        custom_paths = {
            'files': [('lib/libQGLViewer'+addition+'.prl', 'lib64/libQGLViewer'+addition+'.prl'),
                      ('lib/libQGLViewer'+addition+'.%s' % shlib_ext,
                          'lib64/libQGLViewer'+addition+'.%s' % shlib_ext)],
            'dirs': ['include/QGLViewer'],
        }
```

**Review by Micket — CHANGES_REQUESTED** [review]

File: easybuild/easyblocks/l/libqglviewer.py [hunk 3]
```diff
@@ -49,10 +52,25 @@ def configure_step(self):
...
         """Custom sanity check for libQGLViewer."""
         shlib_ext = get_shared_lib_ext()
-
+        """
+        From version 2.8.0 onwards qt version also gets added to the lib file names.
+        """
+        if LooseVersion(self.version) < LooseVersion("2.8.0"):
+            suffix = ''
+        else:
+            suffix = ''
```
Comment (Micket) [inline easybuild/easyblocks/l/libqglviewer.py:61]:
```suggestion
```

File: easybuild/easyblocks/l/libqglviewer.py [hunk 4]
```diff
@@ -49,10 +52,25 @@ def configure_step(self):
...
+            for dep in ['Qt5', 'Qt6']:
+                if get_software_root(dep):
+                    suffix = dep.lower()
+                    suffix = '-' + suffix
```
Comment (Micket) [inline easybuild/easyblocks/l/libqglviewer.py:65]:
```suggestion
                    suffix = '-' + dep.lower()
```

**Review by Micket — CHANGES_REQUESTED** [review]

**Review by Micket — APPROVED** [review]
lgtm

**Comment by Micket** [comment]
I need a test report before i can merge this. Do you have a PR for a newer libQGLViewer?

**Comment by satishskamath** [comment]
> I need a test report before i can merge this. Do you have a PR for a newer libQGLViewer?
https://github.com/easybuilders/easybuild-easyconfigs/pull/15945 , I just created a PR for the easyconfig. But the tests will fail without this PR. :)

**Pull request merged**